compiler/rustc_mir_build/src/thir/pattern/_match.rs

```diff
@@ -1485,9 +1485,7 @@ fn all_constructors<'a, 'tcx>(
         )
     };
     match *pcx.ty.kind() {
-        ty::Bool => {
-            [true, false].iter().map(|&b| ConstantValue(ty::Const::from_bool(cx.tcx, b))).collect()
-        }
+        ty::Bool => vec![make_range(0, 1)],
         ty::Array(ref sub_ty, len) if len.try_eval_usize(cx.tcx, cx.param_env).is_some() => {
```

Comment:
Doesn't this affect diagnostics? I'd expect this would display "pattern `0` not covered" instead of "pattern `false` not covered".

Reply:
I had also wondered this, but assumed this would be caught by tests, so it must have been okay. But you're right that it would be better to understand this (and add a comment).

Reply:
This is because while *decisions* are made on `Constructor`s, they do not directly flow into diagnostics. Instead diagnostics are reported on `Pat`s in https://github.com/rust-lang/rust/blob/6553d38bcb0c9193808ae93db8bd95bcf4c4824a/compiler/rustc_mir_build/src/thir/pattern/check_match.rs